furusiya/function_mess.py

```diff
@@ -216,10 +219,20 @@ def play_game():
         Game.renderer.render()
 
         # handle keys and exit game if needed
-        player_action = handle_keys()
-        if player_action == 'exit':
-            save_game()
-            break
+        if (isinstance(player_action, dict)
+                and player_action.get('turnsToFullyRest', 0) > 0
+                and not [
+                    e
+                    for e in Game.area_map.entities
+                    if e.hostile and (e.x, e.y) in Game.renderer.visible_tiles
+                ]):
+            player_action['turnsToFullyRest'] -= 1
```

Comment:
I don't understand this line. What does it actually do?

Are we resting first, and restoring full HP, and then iterating and actually passing time turn by turn until something hostile appears?

Does that mean that if an enemy is just barely out of sight, and I rest to full (which takes ten turns), and they appear the very first turn, I still have full HP?

Reply:
Nope. It works as expected.

            player_action['turnsToFullyRest'] -= 1
            Game.player.rest()

It rests for the number of turns needed to reach full health.